crossbeam-epoch/src/sync/list.rs

```diff
@@ -243,36 +243,43 @@ impl<'g, T: 'g, C: IsElement<T>> Iterator for Iter<'g, T, C> {
                 // This entry was removed. Try unlinking it from the list.
                 let succ = succ.with_tag(0);
 
-                // The tag should never be zero, because removing a node after a logically deleted
+                // The tag should always be zero, because removing a node after a logically deleted
                 // node leaves the list in an invalid state.
                 debug_assert!(self.curr.tag() == 0);
 
-                match self
+                // Try to unlink `curr` from the list, and get the new value of `self.pred`.
+                let succ = self
```

Comment:
Can you perhaps rewrite this as `let succ = match self.pred.compare_and_set...`?

I find matching easier to understand than combinators like `.map(...).unwrap_or_else(...)` -- I have to pause and think for a moment what is actually going on here.

Reply:
I agree.  I changed it and rebased on `master`.  Thanks~